src/Lexer.ts

```diff
@@ -0,0 +1,98 @@
+// @author Anush kumar N <anush.kumar@codingmart.com>
+
+import TokenType from './Token';
+
+type arrayOfString = Array<string>;
+type literalWithType = { type: Symbol; literal: string | undefined };
+type identifyLiteralReturnType = { token: literalWithType; updatedPos: number | undefined };
```

Comment:
These type aliases are quite compact. Can we give them some room to breath and spread them across multiple lines? i.e.

```
type literalWithType = {
  type: Symbol;
  literal: string;
};

Reply:
The [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) itself urges readers to prefer interfaces over type aliases.

Choice quote:
> For the most part, you can choose based on personal preference, and TypeScript will tell you if it needs something to be the other kind of declaration. If you would like a heuristic, use interface until you need to use features from type.

Let's follow this advice?

Reply:
yes sure, i have also reduced the number of types or interfaces that we are using making the code more easy to understand and look nice